### PR TITLE
Adding m17-cxx-demod, program to modulate and demodulate data using the M17 protocol

### PR DESCRIPTION
**Pull request by n1ai**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by ryanvolz** [comment]
I confirm that I am willing to be a maintainer.

File: recipes/m17-cxx-demod/bld.bat [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+mkdir build
+cd build
+
+:: configure
+cmake -G "Ninja" ^
+    -DCMAKE_BUILD_TYPE:STRING=Release ^
+    -DCMAKE_INSTALL_PREFIX:PATH="%LIBRARY_PREFIX%" ^
+    -DCMAKE_PREFIX_PATH:PATH="%LIBRARY_PREFIX%" ^
+    -DENABLE_DOXYGEN=OFF ^
+    -DENABLE_TESTING=ON ^
```
Comment (ryanvolz) [inline recipes/m17-cxx-demod/bld.bat:14]:
```suggestion

```
This package does nothing with these flags (particular to GNU Radio usually), so you can remove them.

File: recipes/m17-cxx-demod/build.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+mkdir build
+cd build
+
+cmake_config_args=(
+    -DCMAKE_BUILD_TYPE=Release
+    -DCMAKE_INSTALL_PREFIX=$PREFIX
+    -DLIB_SUFFIX=""
+    -DENABLE_DOXYGEN=OFF
+    -DENABLE_TESTING=ON
```
Comment (ryanvolz) [inline recipes/m17-cxx-demod/build.sh:13]:
```suggestion

```
Same here, remove.

File: recipes/m17-cxx-demod/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cmake >=3.9
+    - make  # [not win]
+    - ninja  # [win]
+    - pkg-config  # [not win]
```
Comment (ryanvolz) [inline recipes/m17-cxx-demod/meta.yaml:22]:
```suggestion
    - ninja
    - pkg-config
```
Switch to using `ninja` and `pkg-config` everywhere and the builds should at least proceed further.

File: recipes/m17-cxx-demod/bld.bat (hunk 1 above)
Comment (n1ai) [reply recipes/m17-cxx-demod/bld.bat]: Accepted.

File: recipes/m17-cxx-demod/build.sh (hunk 2 above)
Comment (n1ai) [reply recipes/m17-cxx-demod/build.sh]: Accepted.

File: recipes/m17-cxx-demod/meta.yaml (hunk 3 above)
Comment (n1ai) [reply recipes/m17-cxx-demod/meta.yaml]: Accepted.

**Comment by ryanvolz** [comment]
I've definitely encountered both of the current Windows build problems before: use of "and" and "or" as logic keywords and something to do with "popcnt". Now I have to figure out where I've seen them and what the solution is.

**Comment by ryanvolz** [comment]
@conda-forge/staged-recipes Ready for review. Thanks!

**Pull request merged**